### PR TITLE
applications: nrf5340_audio: bt_stream documentation

### PR DESCRIPTION
**Pull request by erikrobstad**
Made the necessary changes to the doc
to fit the adding of bt_stream.

### CODE REVIEW

File: doc/nrf/releases_and_maturity/software_maturity.rst [hunk 1]
```diff
@@ -148,21 +148,21 @@ The following table indicates the software maturity levels of the support for th
           * BAP unicast, one CIG with two CIS.
```
Comment (alexsven) [inline doc/nrf/releases_and_maturity/software_maturity.rst]: BAP was removed for the rest, so it should probably be removed here as well?

**Comment by greg-fer** [comment]
Relates to #32

**Review by alexsven — APPROVED** [review]

File: doc/nrf/releases_and_maturity/software_maturity.rst (hunk 1 above)
Comment (erikrobstad) [reply doc/nrf/releases_and_maturity/software_maturity.rst]: The reason why it was not removed here is that we have not implemented CAP for unicast client. But I don't mind if we remove it

File: doc/nrf/images/octave_application_structure.svg
Comment (erikrobstad) [inline doc/nrf/images/octave_application_structure.svg:1]: For the four streamctrl files it should be written "One of the following:"

File: doc/nrf/images/octave_application_structure_gateway.svg
Comment (erikrobstad) [inline doc/nrf/images/octave_application_structure_gateway.svg:1]: For the four streamctrl files it should be written "One of the following:"

File: doc/nrf/images/octave_application_structure_sync_module.svg
Comment (erikrobstad) [inline doc/nrf/images/octave_application_structure_sync_module.svg:1]: For the four streamctrl files it should be written "One of the following:"

**Review by erikrobstad — COMMENTED** [review]
@greg-fer I made some minor comments

File: doc/nrf/images/octave_application_structure_gateway.svg
Comment (greg-fer) [reply doc/nrf/images/octave_application_structure_gateway.svg:1]: There are just two files here. Should I include all four?

Comment (erikrobstad) [reply doc/nrf/images/octave_application_structure_gateway.svg:1]: No it is correct to only have the two files

**Review by greg-fer — APPROVED** [review]

File: applications/nrf5340_audio/doc/firmware_architecture.rst [hunk 2]
```diff
@@ -141,13 +148,15 @@ Communication between modules is primarily done through Zephyr's :ref:`zephyr:zb
 
 The application uses the following buses:
 
-  * ``le_audio_chan`` - For handling LE Audio events from :file:`le_audio_cis_headset.c`, :file:`le_audio_bis_headset.c`, :file:`le_audio_cis_gateway.c`, or :file:`le_audio_bis_gateway.c`.
+  * ``le_audio_chan`` - For handling LE Audio events from the Bluetooth stream modules, specifically :file:`unicast_client.c`, :file:`unicast_server.c`, :file:`broadcast_source.c`, and :file:`broadcast_sink.c`.
```
Comment (koffes) [inline applications/nrf5340_audio/doc/firmware_architecture.rst:151]: IMO, this is not something we should document:
- It is quite technical (i.e. for developers)
- It is quite well documented in the code
- There is overhead to maintain this doc.

Comment (koffes) [reply applications/nrf5340_audio/doc/firmware_architecture.rst:151]: No action requested from me, just my two cents :)

File: doc/nrf/releases_and_maturity/software_maturity.rst [hunk 3]
```diff
@@ -123,9 +123,9 @@ The following table indicates the software maturity levels of the support for th
           Play and pause emulated by disabling and enabling stream, respectively.
         - The following limitations apply:
 
-          * Basic Audio Profile (BAP) broadcast, one BIG with two BIS streams.
+          * One BIG with two BIS streams.
           * Audio input: USB or I2S (Line in or using Pulse Density Modulation).
-          * Configuration: 16 bit, several bit rates ranging from 24 kbps to 160 kbps.
+          * Configuration: 16 or 32 bit, several bit rates ranging from 32 kbps to 124 kbps.
```
Comment (koffes) [inline doc/nrf/releases_and_maturity/software_maturity.rst]: Can you double check CI coverage on this?

Comment (koffes) [reply doc/nrf/releases_and_maturity/software_maturity.rst]: Goes for all 32 bit mentions.

Comment (koffes) [reply doc/nrf/releases_and_maturity/software_maturity.rst]: Not sure 32 bit should be mentioned until we have discussed. Some modules do not support 32 bit.

**Review by koffes — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/software_maturity.rst (hunk 3 above)
Comment (erikrobstad) [reply doc/nrf/releases_and_maturity/software_maturity.rst]: Removed 32 bit for now

**Comment by erikrobstad** [comment]
Added changelog and rebased

**Review by koffes — APPROVED** [review]

**Pull request merged**